src/users/UserList.js

```diff
@@ -22,48 +22,46 @@ type Props = $ReadOnly<{|
   onPress: (user: UserOrBot) => void,
 |}>;
 
-export default class UserList extends PureComponent<Props> {
-  static defaultProps = {
-    selected: [],
-  };
+export default function UserList(props: Props) {
+  const { filter, users, presences, onPress, selected } = props;
+  const shownUsers = sortUserList(filterUserList(users, filter), presences);
 
-  render() {
-    const { filter, users, presences, onPress, selected } = this.props;
-    const shownUsers = sortUserList(filterUserList(users, filter), presences);
-
-    if (shownUsers.length === 0) {
-      return <SearchEmptyState text="No users found" />;
-    }
+  if (shownUsers.length === 0) {
+    return <SearchEmptyState text="No users found" />;
+  }
 
-    const groupedUsers = groupUsersByStatus(shownUsers, presences);
-    const sections = Object.keys(groupedUsers).map(key => ({
-      key: `${key.charAt(0).toUpperCase()}${key.slice(1)}`,
-      data: groupedUsers[key].map(u => u.user_id),
-    }));
+  const groupedUsers = groupUsersByStatus(shownUsers, presences);
+  const sections = Object.keys(groupedUsers).map(key => ({
+    key: `${key.charAt(0).toUpperCase()}${key.slice(1)}`,
+    data: groupedUsers[key].map(u => u.user_id),
+  }));
 
-    return (
-      <SectionList
-        style={styles.list}
-        stickySectionHeadersEnabled
-        keyboardShouldPersistTaps="always"
-        initialNumToRender={20}
-        sections={sections}
-        keyExtractor={item => item}
-        renderItem={({ item }) => (
-          <UserItem
-            key={item}
-            userId={item}
-            onPress={onPress}
-            isSelected={!!selected.find(user => user.user_id === item)}
-          />
-        )}
-        renderSectionHeader={({ section }) =>
-          section.data.length === 0 ? null : (
-            // $FlowFixMe[incompatible-type]
-            <SectionHeader text={section.key} />
-          )
-        }
-      />
-    );
-  }
+  return (
+    <SectionList
+      style={styles.list}
+      stickySectionHeadersEnabled
+      keyboardShouldPersistTaps="always"
+      initialNumToRender={20}
+      sections={sections}
+      keyExtractor={item => item}
+      renderItem={({ item }) => (
+        <UserItem
+          key={item}
+          userId={item}
+          onPress={onPress}
+          isSelected={!!selected.find(user => user.user_id === item)}
+        />
+      )}
+      renderSectionHeader={({ section }) =>
+        section.data.length === 0 ? null : (
+          // $FlowFixMe[incompatible-type]
+          <SectionHeader text={section.key} />
+        )
+      }
+    />
+  );
 }
+
+UserList.defaultProps = {
+  selected: [],
```

Comment:
Don't use `defaultProps` for a functional component; instead use plain old JS destructuring:
```js
export default function UserList(props: Props) {
  const { foo, bar, selected = [] } = props;
```

The main reason is that it keeps the default visible in a natural spot at the top of the function.

This also means the type in `Props` gets to start being optional, reflecting the API actually exposed to the caller.

Reply:
More on this here:
https://github.com/zulip/zulip-mobile/pull/4440#discussion_r565684617

and I guess I should go add it to the style guide.

Reply:
https://github.com/zulip/zulip-mobile/blob/master/docs/style.md#react-function-prop-defaults

Reply:
Thanks for pointing that out! I had a feeling there was a better way of doing this, but didn't know what it was :)